python/ngen_conf/src/ngen/config/init_config/lstm.py

```diff
@@ -0,0 +1,123 @@
+from ngen.init_config import serializer_deserializer as serde
+from pathlib import Path
+
+from typing import Optional, Literal
+
+
+# TODO: add tests
+# TODO: modify once a better understanding of how LSTM is to be configured in practice. Based on the
+# configs present in https://github.com/NOAA-OWP/lstm it is unclear when certain keys are required.
+class LSTM(serde.YamlSerializerDeserializer):
+    train_cfg_file: Path
+
+    # TODO: not sure what other values this can be
+    initial_state: Literal["zero"] = "zero"
+    verbose: bool = False
+    basin_name: Optional[str] = None
+    basin_id: Optional[str] = None
+
+    # NOTE: listed as post-processing / plotting only
+    # lat: float
+    # lon: float
+    area_sqkm: float
```

Comment:
My understanding of the LSTM is that depending on which `train_cfg_file` is used, the required parameters can be different...

Reply:
That is my understand as well. I don't know all the possible combinations of parameters, so if we want to leave `LSTM` out of this PR and add it at a later date, im fine with that. Thoughts?

Reply:
I don't know the context here, but I would like to. In any rate, why is LSTM configuration files needed in a calibration repository anyway? LSTM would be trained via back propogation, no calibration necessary. In terms of the required parameters, it pretty much comes down to what would be available. It became clear that the majority of the CAMELS attributes would not be available in the hydrofabric, so we trained on only a couple attributes we knew would be available (slope and elevation, etc.). However, it appears there are several NOAH-MP parameters that are available in the hydrofabric, so I think the best thing to do would be to train an LSTM on those. Which I am planning to do soon-ish.

Reply:
because this repo no longer just does calibration things, there is a configuration package developed in this repo that is used by the calibration, but also by other workflows that need to generate configuration files and components programatically.

Reply:
Welll @jmframe , the repo name `ngen-cal` is not the most descriptive. This repo has become a catchall for `ngen` configuration _and_ calibration related concerns. The goal of the `ngen.init_config` package is to capture the schema for `bmi` module configuration files (e.g. `LSTM`). This is desirable so we can fully validate an `ngen` realization config and all linked `bmi` config files.

With that in mind, it would be extremely helpful to know _which_ parameters _can_ appear in a config file and _when_ (i.e. if that is conditional).

Reply:
Okay, I see. That makes sense. I think that a re-visit to the LSTM module with the maturation of NGen and Hydrofabric over the past two years is warranted. It would be great to have a chat with @mikejohnson51 and get an update on what static catchment attributes are always going to be available. Then re-train a few models and do some experimenting with this in mind. Then make a decision on the best keys to include. But for now, the LSTM configuration should have at least: lat, lon, area_sqkm, elev_mean, slope_mean

Reply:
Thanks for the input, @jmframe! For now, I will only include the keys you mentioned and in the future if things change, we can add others.

Reply:
Let's open an issue to come back to LSTM configs and see if we can use a target `train_cfg_file` to determine fields required to support that particular model.  Having a "default" for now is a good place to start.